angular/projects/admin-nrpti/src/app/mines/mines-detail/mines-detail.component.ts

```diff
@@ -15,55 +13,115 @@ import { FactoryService } from '../../services/factory.service';
 export class MinesDetailComponent implements OnInit, OnDestroy {
   private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
 
-  public record = null;
+  public mine = null;
+  public canPublish = false;
+  public isPublished = false;
 
   constructor(public route: ActivatedRoute,
               public router: Router,
               private factoryService: FactoryService,
-              public changeDetectionRef: ChangeDetectorRef,
-              private dialogService: DialogService) {
+              public changeDetectionRef: ChangeDetectorRef) {
   }
 
   ngOnInit() {
     this.route.data.pipe(takeUntil(this.ngUnsubscribe)).subscribe((res: any) => {
-      if (!res || !res.record) {
+      if (!res || !res.mine) {
         alert("Uh-oh, couldn't load Mine");
         this.router.navigate(['mines']);
         return;
       }
 
-      this.record = res.record[0] && res.record[0].data && new Mine(res.record[0].data);
+      this.mine = res.mine[0] && res.mine[0].data && new Mine(res.mine[0].data);
+
+      this.isPublished = this.isRecordPublished();
+      this.canPublish = this.checkCanPublish();
 
       this.changeDetectionRef.detectChanges();
     });
   }
 
-  delete() {
-    this.dialogService.addDialog(ConfirmComponent,
-    {
-      title: 'Confirm Deletion',
-      message: 'Do you really want to delete this Mine Item?',
-      okOnly: false
-    }, {
-      backdropColor: 'rgba(0, 0, 0, 0.5)'
-    })
-    .pipe(takeUntil(this.ngUnsubscribe))
-    .subscribe(
-      isConfirmed => {
-        if (isConfirmed) {
-          try {
-            this.factoryService.deleteMineItem(this.record._id, 'mine');
-            this.router.navigate(['mines']);
-          } catch (e) {
-            alert('Could not delete Mine Item');
-          }
+  isRecordPublished(): boolean {
+    return this.mine && this.mine.read && this.mine.read.includes('public');
+  }
+
+  checkCanPublish(): boolean {
+    return (this.mine.name
+      && this.mine.status
+      && this.mine.permitNumbers
+      && this.mine.permitNumbers.length > 0
+      && this.mine.tailingsImpoundments
+      && this.mine.commodities
+      && this.mine.commodities.length > 0
+      && this.mine.location
+      && this.mine.location.coordinates
+      && this.mine.location.coordinates.length > 0);
+  }
+
+  publish(): void {
+    this.factoryService
+      .publishRecord(this.mine)
+      .pipe(
```

Comment:
I don't think there is any need for `.pipe` here. An error can be handled in subscribe, so there is not need to combine it with `.takeUntil`. I am pretty sure this could be written something like:

`this.factoryService
      .publishRecord(this.Mine)
      .takeUntil(this.ngUnsubscribe)
      . subscribe(
           // Place your main function here,
           // Place your error handler here in the second position
       );`

Reply:
I'll try it out, thanks

Reply:
Not an RXJS expert, so could be wrong:

The pipe catchError is there to catch errors that happen outside of the observable chain. So in this case, if `unPublishRecord()` throws an error for any reason prior to returning an observable. 

I think current RXJS requires pipe for all of its modifier type addons, like takeUntil?

Reply:
@NickPhura is correct. I am not used to that, but anything version 6 or greater uses modifiers in pipes now.

Reply:
I left them as is, thanks guys